tools/hidl/hidlcomp.cpp

```diff
@@ -4160,6 +4165,17 @@ void EspServInfo::write_esp_binding(const char *packagename)
     }
     outs("}\n");
 
+    // init_maps implementation
+    outf("\nvoid C%sSoapBinding::init_maps()\n", name_);
+    outs("{\n");
+    for (mthi=methods; mthi!= nullptr; mthi=mthi->next)
+    {
+        std::string methodNameKey(mthi->getName());
+        std::transform(methodNameKey.cbegin(), methodNameKey.cend(), methodNameKey.begin(), tolower);
```

Comment:
A helper function could also be added to make this code cleaner/simpler.

outf("\taddMethod(\"%s\");\n", mthi->getName());

Even better would be

CHttpSoapBinding::registerMethods(std::initializer_list<const char *> methods)
...
init_maps()
{
   registerMethods({"a","b","c","d"});
}